#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9999813443775041 -2.87212628977389e-06 0.00610826396862956 1.4881024266313612
  position -0.268255057873225 3.155538602258725 0.4412031783241407
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.24 0 0
  floorSize 2.8 1.6
  floorTileSize 2.8 1.6
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "Arena_LineFollower.png"
      ]
    }
    roughness 1
    roughnessMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
  }
  wallHeight 0.01
}
Robot {
  translation -0.620138 -0.00509102 0.412406
  rotation 3.137019997134908e-05 -0.9999999990866835 -2.9026599973489526e-05 0.0374154
  children [
    DEF body Transform {
      translation 0 0.03 0
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 1 0 0
          }
          geometry Box {
            size 0.15 0.025 0.1
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 755.8917227952123
        axis 0 0 1
        anchor 0.04 0.03 -0.054
      }
      device [
        RotationalMotor {
          name "front_right_motor"
        }
      ]
      endPoint Solid {
        translation 0.04 0.03 -0.054
        rotation -0.4476488013932164 -0.6323012475540173 0.632301892257844 2.299797997054969
        children [
          DEF wheel Shape {
            appearance Asphalt {
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 770.3962854063323
        axis 0 0 1
        anchor 0.04 0.03 0.054
      }
      device [
        RotationalMotor {
          name "front_left_motor"
        }
      ]
      endPoint Solid {
        translation 0.04 0.03 0.054
        rotation 0.2521367475850648 -0.6842609537751834 0.6842616514579065 3.6355705498042843
        children [
          DEF wheel Shape {
            appearance Asphalt {
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 773.9122777855212
        axis 0 0 1
        anchor -0.04 0.03 0.054
      }
      device [
        RotationalMotor {
          name "back_left_motor"
        }
      ]
      endPoint Solid {
        translation -0.04 0.03 0.054
        rotation 0.7626207375207975 0.45738888491474705 -0.45738935127490604 4.444652779289389
        children [
          DEF wheel Shape {
            appearance Asphalt {
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 755.8916695857971
        axis 0 0 1
        anchor -0.04 0.03 -0.054
      }
      device [
        RotationalMotor {
          name "back_right_motor"
        }
      ]
      endPoint Solid {
        translation -0.04 0.03 -0.054
        rotation -0.4476689946152128 -0.6322940992892341 0.6322947439795944 2.2997643528417857
        children [
          DEF wheel Shape {
            appearance Asphalt {
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    Solid {
      translation -0.075 0.03 -0.03
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "solid(4)"
    }
    Solid {
      translation -0.075 0.03 0
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "solid(5)"
    }
    Solid {
      translation -0.075 0.03 0.03
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "solid(6)"
    }
    Solid {
      translation -0.04 0.0295 -0.055
      rotation 1 0 0 1.5708
      children [
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.015
          }
        }
      ]
      name "solid(7)"
    }
    Solid {
      translation 0.04 0.0295 -0.055
      rotation 1 0 0 1.5708
      children [
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.015
          }
        }
      ]
      name "solid(8)"
    }
    Solid {
      translation 0.04 0.0294999 0.055
      rotation 1 0 0 1.5708
      children [
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.015
          }
        }
      ]
      name "solid(9)"
    }
    Solid {
      translation -0.04 0.0294999 0.055
      rotation 1 0 0 1.5708
      children [
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.015
          }
        }
      ]
      name "solid(10)"
    }
    Solid {
      translation 0.06 0.04 -0.03
      rotation -1 3.59841e-07 2.09385e-07 1.5708
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(11)"
    }
    Solid {
      translation 0.053 0.04 -0.03
      rotation 1 -6.30423e-07 -3.71658e-07 -1.5708053071795867
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(12)"
    }
    Solid {
      translation 0.058 0.04 -0.03
      rotation 0.9831055672050736 -0.12942994302074512 -0.12942694302206584 -1.5878453071795864
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(13)"
    }
    Solid {
      translation 0.055 0.04 -0.03
      rotation 0.9831059489305966 0.12942699327665613 0.12942699327665613 -1.5878453071795864
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(14)"
    }
    Solid {
      translation 0.048 0.04 -0.03
      rotation 0.9831059489305966 0.12942699327665613 0.12942699327665613 -1.5878453071795864
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(15)"
    }
    Solid {
      translation 0.045 0.04 -0.03
      rotation 0.9831055672050736 -0.12942994302074512 -0.12942694302206584 -1.5878453071795864
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(16)"
    }
    Solid {
      translation 0.038 0.04 -0.03
      rotation -0.9351132545406678 0.2505650682045725 0.25056006820321153 1.63784
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(17)"
    }
    Solid {
      translation 0.038 0.04 -0.03
      rotation 0.9351138487168261 0.25056095946412593 0.2505619594639642 -1.6378453071795862
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(18)"
    }
    Solid {
      translation 0.071 0.031 0.04
      rotation 2.184229999990977e-06 0.9999999999958691 -1.8684599999922816e-06 -1.5707953071795862
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(19)"
    }
    Solid {
      translation 0.071 0.031 0.033
      rotation 2.184229999990977e-06 0.9999999999958691 -1.8684599999922816e-06 -1.5707953071795862
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(20)"
    }
    Solid {
      translation 0.071 0.031 0.027
      rotation 2.184229999990977e-06 0.9999999999958691 -1.8684599999922816e-06 -1.5707953071795862
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(21)"
    }
    Solid {
      translation 0.071 0.0255 0.0370001
      rotation -0.57735126918836 -0.5773482691869614 0.57735126918836 2.0944
      children [
        Shape {
          geometry Box {
            size 0.001 0.007 0.01
          }
        }
      ]
      name "solid(22)"
    }
    Solid {
      translation 0.071 0.0366 0.024
      rotation -0.57735126918836 -0.5773482691869614 0.57735126918836 2.0944
      children [
        Shape {
          geometry Box {
            size 0.001 0.007 0.01
          }
        }
      ]
      name "solid(23)"
    }
    Solid {
      translation 0.071 0.033 0.021
      rotation -0.7071072811853786 1.8354307298698489e-06 0.7071062811849809 -3.1415853071795863
      children [
        Shape {
          geometry Box {
            size 0.001 0.007 0.01
          }
        }
      ]
      name "solid(24)"
    }
    Solid {
      translation 0.071 0.03 0.0240001
      rotation -0.57735126918836 -0.5773482691869614 0.57735126918836 2.0944
      children [
        Shape {
          geometry Box {
            size 0.001 0.007 0.01
          }
        }
      ]
      name "solid(25)"
    }
    Solid {
      translation 0.071 0.031 0.01
      rotation -0.12943002000826476 -0.9831051519757795 0.12943002000826476 1.58783
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(26)"
    }
    Solid {
      translation 0.071 0.0364 0.012
      rotation -0.57735126918836 -0.5773482691869614 0.57735126918836 2.0944
      children [
        Shape {
          geometry Box {
            size 0.001 0.007 0.01
          }
        }
      ]
      name "solid(27)"
    }
    Solid {
      translation 0.071 0.031 0.004
      rotation 2.184229999990977e-06 0.9999999999958691 -1.8684599999922816e-06 -1.5707953071795862
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(28)"
    }
    Solid {
      translation 0.071 0.031 -0.003
      rotation 2.184229999990977e-06 0.9999999999958691 -1.8684599999922816e-06 -1.5707953071795862
      children [
        Shape {
          geometry Box {
            size 0.001 0.012 0.01
          }
        }
      ]
      name "solid(29)"
    }
    Solid {
      translation 0.071 0.0254996 0
      rotation -0.57735126918836 -0.5773482691869614 0.57735126918836 2.0944
      children [
        Shape {
          geometry Box {
            size 0.001 0.007 0.01
          }
        }
      ]
      name "solid(30)"
    }
    Solid {
      translation 0.071 0.0365 -6.29174e-08
      rotation -0.57735126918836 -0.5773482691869614 0.57735126918836 2.0944
      children [
        Shape {
          geometry Box {
            size 0.001 0.007 0.01
          }
        }
      ]
      name "solid(31)"
    }
    DistanceSensor {
      translation -0.075 0.026 0
      rotation 0 0 1 -2.6179953071795863
      name "middle_ir"
      type "infra-red"
      numberOfRays 20
      aperture 0.2
    }
    DistanceSensor {
      translation -0.075 0.026 0.03
      rotation 0.27728098300028475 0.07429649544498416 0.9579119412717378 -2.1140253071795865
      name "left_ir"
      type "infra-red"
      numberOfRays 30
      aperture 0.4
    }
    DistanceSensor {
      translation -0.075 0.026 -0.03
      rotation 0.2772809665193942 0.07429729102888906 -0.9579118843358396 2.11401
      name "right_ir"
      type "infra-red"
      numberOfRays 30
      aperture 0.4
    }
    DistanceSensor {
      translation 0.035 0.026 0.05
      rotation -0.6947469307028182 0.18615798143176612 -0.6947459307029179 2.77349
      name "stop_sensor"
      type "infra-red"
      numberOfRays 20
      aperture 0.2
    }
    Solid {
      translation -0.04 0.07 0
      rotation 0 0 1 0
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.172702 0.13135 0.969131
          }
          geometry Box {
            size 0.04 0.09 0.1
          }
        }
      ]
      name "max_face"
    }
    Solid {
      translation -0.055 0.1 0.02
      children [
        Shape {
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "solid(33)"
    }
    Solid {
      translation -0.0600959 0.065 0
      rotation 0 0 1 1.57
      children [
        Shape {
          geometry Plane {
            size 0.025 0.05
          }
        }
      ]
      name "solid(37)"
    }
    Solid {
      translation -0.055 0.1 -0.02
      children [
        Shape {
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "solid(34)"
    }
    Solid {
      translation -0.060048 0.100048 0.02
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.118944 0.872526 0.849821
          }
          geometry Cylinder {
            height 0.01
            radius 0.007
          }
        }
      ]
      name "solid(35)"
    }
    Solid {
      translation -0.040072 0.090032 0.059984
      rotation 1 0 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.03
            radius 0.01
          }
        }
      ]
      name "solid(42)"
    }
    Solid {
      translation -0.040072 0.0899361 -0.060016
      rotation 1 0 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.03
            radius 0.01
          }
        }
      ]
      name "solid(43)"
    }
    Solid {
      translation -0.060048 0.100048 -0.02
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.118944 0.872526 0.849821
          }
          geometry Cylinder {
            height 0.01
            radius 0.007
          }
        }
      ]
      name "solid(36)"
    }
    Solid {
      translation -0.06 0.0796593 0.0374118
      rotation 1 0 0 -0.261795307179586
      children [
        Shape {
          geometry Box {
            size 0.001 0.002 0.04
          }
        }
      ]
      name "solid(38)"
    }
    Solid {
      translation -0.06 0.0492944 -0.0372743
      rotation 1 0 0 -0.261795307179586
      children [
        Shape {
          geometry Box {
            size 0.001 0.002 0.04
          }
        }
      ]
      name "solid(41)"
    }
    Solid {
      translation -0.06 0.05 0.0348236
      rotation 1 0 0 0.261799
      children [
        Shape {
          geometry Box {
            size 0.001 0.002 0.04
          }
        }
      ]
      name "solid(39)"
    }
    Solid {
      translation -0.06 0.0803649 -0.0398625
      rotation 1 0 0 0.261799
      children [
        Shape {
          geometry Box {
            size 0.001 0.002 0.04
          }
        }
      ]
      name "solid(40)"
    }
    Solid {
      translation -0.04 0.149944 0.070032
      rotation 1 0 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.011 0.011 0.1
          }
        }
      ]
      name "solid(44)"
    }
    Solid {
      translation -0.04 0.149832 -0.069968
      rotation 1 0 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.011 0.011 0.1
          }
        }
      ]
      name "solid(45)"
    }
    Solid {
      translation -0.040072 0.119984 -3.98561e-05
      rotation 1 0 0 -0.000795307179585869
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 1 0 0
          }
          geometry Cylinder {
            height 0.03
            radius 0.01
          }
        }
      ]
      name "solid(46)"
    }
    Solid {
      translation 0.04 0.0294999 0.056
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.005
          }
        }
      ]
      name "solid(32)"
    }
    Solid {
      translation -0.04 0.0294999 0.056
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.005
          }
        }
      ]
      name "solid(47)"
    }
    Solid {
      translation -0.04 0.0294999 -0.056
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.005
          }
        }
      ]
      name "solid(48)"
    }
    Solid {
      translation 0.04 0.0294999 -0.056
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.005
          }
        }
      ]
      name "solid(49)"
    }
    Solid {
      translation -0.065 0.100016 -0.02
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0 0 0
          }
          geometry Sphere {
            radius 0.003
          }
        }
      ]
      name "solid(50)"
    }
    Solid {
      translation -0.065 0.100016 0.02
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0 0 0
          }
          geometry Sphere {
            radius 0.003
          }
        }
      ]
      name "solid(51)"
    }
    Solid {
      translation 0.07 0.023 -0.037
      rotation 0 1 0 1.5708
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance Asphalt {
          }
          geometry Mesh {
            url [
              "meshes/suzanne.obj"
            ]
          }
        }
      ]
      name "solid(52)"
    }
    Solid {
      translation 0.0106814 0.0551764 0.02
      rotation 0 0 1 -0.523595307179586
      children [
        Shape {
          appearance SolarCell {
          }
          geometry Plane {
            size 0.07 0.05
          }
        }
      ]
      name "solid(53)"
    }
    Solid {
      translation 0.0506814 0.0451764 -0.01
      rotation 0 0 1 4.07987e-06
      children [
        Shape {
          appearance Pcb {
          }
          geometry Plane {
            size 0.03 0.02
          }
        }
      ]
      name "solid(54)"
    }
    Solid {
      translation 0.06 0.06 0.04
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0 0 0
          }
          geometry Cylinder {
            height 0.03
            radius 0.008
          }
        }
      ]
      name "solid(55)"
    }
    Solid {
      translation 0.06 0.08 0.04
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.990417 0.500267 0.0328679
          }
          geometry Cylinder {
            height 0.01
            radius 0.008
          }
        }
      ]
      name "solid(56)"
    }
    Solid {
      translation 0.06 0.081 0.04
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.402777 0.402777 0.402777
          }
          geometry Cylinder {
            height 0.01
            radius 0.006
          }
        }
      ]
      name "solid(57)"
    }
    Solid {
      translation 0.06 0.082 0.04
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.970001 0.970001 0.970001
          }
          geometry Cylinder {
            height 0.01
            radius 0.001
          }
        }
      ]
      name "solid(58)"
    }
    Solid {
      translation 0.06 0.06 0.02
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0 0 0
          }
          geometry Cylinder {
            height 0.03
            radius 0.008
          }
        }
      ]
      name "solid(59)"
    }
    Solid {
      translation 0.06 0.08 0.02
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.990417 0.500267 0.0328679
          }
          geometry Cylinder {
            height 0.01
            radius 0.008
          }
        }
      ]
      name "solid(60)"
    }
    Solid {
      translation 0.06 0.081 0.02
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.402777 0.402777 0.402777
          }
          geometry Cylinder {
            height 0.01
            radius 0.006
          }
        }
      ]
      name "solid(61)"
    }
    Solid {
      translation 0.06 0.082 0.02
      children [
        Shape {
          appearance GlossyPaint {
            baseColor 0.970001 0.970001 0.970001
          }
          geometry Cylinder {
            height 0.01
            radius 0.001
          }
        }
      ]
      name "solid(62)"
    }
    LED {
      translation -0.04 0.21 0.07
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.015
          }
        }
        PointLight {
          attenuation 0 0 1
          color 1 0 0
          intensity 0
          location -0.04 0.21 0.07
          radius 0.3
        }
      ]
    }
    LED {
      translation -0.04 0.21 -0.07
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.015
          }
        }
        PointLight {
          attenuation 0 0 1
          color 1 0 0
          intensity 5
          location -0.04 0.21 0.07
          radius 0.3
        }
      ]
      name "led(1)"
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "max_controller_final"
}
